library/src/main/java/com/devbrackets/android/exomedia/AudioPlayer.kt

```diff
@@ -0,0 +1,447 @@
+     * `trackType`
+     *
+     * @param trackType The type for the track to switch to the selected index
+     * @param groupIndex The index for the group in the [TrackGroupArray] specified by the `trackType`
+     * @param trackIndex The index for the track to switch to
+     */
+    fun setTrack(trackType: ExoMedia.RendererType, groupIndex: Int, trackIndex: Int) {
+        audioPlayerImpl.setTrack(trackType, groupIndex, trackIndex)
+    }
+
+
+    fun getSelectedTrackIndex(type: ExoMedia.RendererType, groupIndex: Int): Int {
+        return audioPlayerImpl.getSelectedTrackIndex(type, groupIndex)
+    }
+
+    /**
+     * Sets the repeatListener to inform of VideoPlayer prepared events
+     *
+     * @param listener The repeatListener
+     */
+    fun setOnPreparedListener(listener: OnPreparedListener?) {
+        listenerMux.setOnPreparedListener(listener)
+    }
+
+    /**
+     * Sets the repeatListener to inform of VideoPlayer completion events
+     *
+     * @param listener The repeatListener
+     */
+    fun setOnCompletionListener(listener: OnCompletionListener?) {
+        listenerMux.setOnCompletionListener(listener)
+    }
+
+    /**
+     * Sets the repeatListener to inform of VideoPlayer buffer update events
+     *
+     * @param listener The repeatListener
+     */
+    fun setOnBufferUpdateListener(listener: OnBufferUpdateListener?) {
+        listenerMux.setOnBufferUpdateListener(listener)
+    }
+
+    /**
+     * Sets the repeatListener to inform of VideoPlayer seek completion events
+     *
+     * @param listener The repeatListener
+     */
+    fun setOnSeekCompletionListener(listener: OnSeekCompletionListener?) {
+        listenerMux.setOnSeekCompletionListener(listener)
+    }
+
+    /**
+     * Sets the repeatListener to inform of playback errors
+     *
+     * @param listener The repeatListener
+     */
+    fun setOnErrorListener(listener: OnErrorListener?) {
+        listenerMux.setOnErrorListener(listener)
+    }
+
+    /**
+     * Sets the repeatListener to inform of ID3 metadata updates
+     *
+     * @param listener The repeatListener to inform
+     */
+    fun setMetadataListener(listener: MetadataListener?) {
+        listenerMux.setMetadataListener(listener)
+    }
+
+    /**
+     * Sets the repeatListener to inform of Analytics updates
+     *
+     * @param listener The repeatListener to inform
+     */
+    fun setAnalyticsListener(listener: AnalyticsListener?) {
+        listenerMux.setAnalyticsListener(listener)
+    }
+
+    /**
+     * Performs the functionality to stop the progress polling, and stop any other
+     * procedures from running that we no longer need.
+     */
+    private fun onPlaybackEnded() {
+        pause()
+    }
+
+    private inner class MuxNotifier : ListenerMux.Notifier() {
+        override fun shouldNotifyCompletion(endLeeway: Long): Boolean {
+            val position = currentPosition
+            val duration = duration
```

Comment:
Name shadowing? Perhaps rename to `dur`?

Reply:
Will Update